### ISSUE TITLE
Billboard clamps to high to terrain?

### ISSUE THREAD

**Issue by pjcozzi** [AnalyticalGraphicsInc/cesium#4062]
Open the KML Sandcastle example:
```
http://localhost:8080/Apps/Sandcastle/index.html?src=KML.html&label=All
```
Turn on AGI World Terrain, then zoom to "Nevada Test Site Area 26."  The label is floating way above terrain.  Shouldn't it clamp to the highest resolution loaded?

![image](https://cloud.githubusercontent.com/assets/782098/16469739/950e836a-3e20-11e6-8bd0-44c583990af7.png)

**Comment by mramato** [comment]
I can't reproduce this in master.  @pjcozzi can you confirm and close?

**Comment by pjcozzi** [comment]
I just tried in master and believe I still see the issue:

![image](https://cloud.githubusercontent.com/assets/782098/18144523/0f3736a0-6f95-11e6-80b3-ef3d0a42d5e9.png)

**Comment by mramato** [comment]
Pretty sure I just forgot to turn terrain on when I tested this. whoops.  The problem is actually pretty bad and happens by jumping to most billboards.  Some end up above the ground, some under the ground.

I've also had some problems with disappearing labels (I think they are improperly being placed under ground like the billboards: 

![image](https://cloud.githubusercontent.com/assets/1495196/18406729/ebd52b72-76ce-11e6-8134-01fb23e47586.png)


It's possible this is a different issue, but unlikely.

**Comment by twpayne** [comment]
> I've also had some problems with disappearing labels (I think they are improperly being placed under ground like the billboards:
...

Is this #2694?

**Comment by mramato** [comment]
No, they are two different issues.  This issue is a nasty race condition in terrain clamping which is causing each letter to be placed in the geographically incorrect position instead of on the terrain itself.  #2694 is a draw order issue and requires us treat labels as special so they always get drawn above terrain if their origin is visible (or something similar).

**Comment by bagnell** [comment]
The problem seems to be picking upsampled terrain. A quick check by disabling upsampling looks like it fixes the problem, but requires more testing.
